service/src/main/kotlin/fi/espoo/evaka/messaging/MessageQueries.kt

```diff
@@ -1519,6 +1521,22 @@ fun Database.Transaction.deleteMessages(contentId: MessageContentId) {
         }
         .execute()
 
+    this.createUpdate<Any> {
+            sql(
+                """
+            DELETE FROM message_thread
+            WHERE id IN (
+                SELECT mt.id
+                FROM message_thread mt
+                JOIN message m on mt.id = m.thread_id
+                WHERE m.content_id = ${bind(contentId)}
+                FOR UPDATE 
+            )
```

Comment:
There exists a shorter syntax for the same thing:

```
DELETE FROM message_thread mt
USING message m
WHERE mt.id = m.thread_id AND m.content_id = ${bind(contentId)}
```

Also not sure why you added `FOR UPDATE` here 🤔

Reply:
... and even if you prefer the longer syntax, you can select directly from `message`:

```
DELETE FROM message_thread
WHERE id IN (
    SELECT thread_id
    FROM message m
    WHERE content_id = ${bind(contentId)}
)
```

Reply:
💡